src/validator.cpp

```diff
@@ -679,10 +679,10 @@ void Validator::ValidatorImpl::validateVariable(const VariablePtr &variable, con
         mValidator->addError(err);
     } else if (!isStandardUnitName(variable->units())) {
         ComponentPtr component = std::dynamic_pointer_cast<Component>(variable->parent());
-        ModelPtr model = std::dynamic_pointer_cast<Model>(component->parent());
+        ModelPtr model = owningModel(component);
         if ((model != nullptr) && !model->hasUnits(variable->units())) {
             ErrorPtr err = std::make_shared<Error>();
-            err->setDescription("Variable '" + variable->name() + "' has an invalid units reference '" + variable->units() + "' that does not correspond with a standard unit or units in the variable's parent component or model.");
+            err->setDescription("Variable '" + variable->name() + "' has an units reference '" + variable->units() + "' that does not correspond with a standard units and is not a units defined in the variable's model.");
```

Comment:
"... a units..." rather than "... an units..."

Reply:
I find the use of "units" as a singular term really clumsy ... maybe we can make it clearer using "unit" and "compound unit" to make the distinction between the `unit` and `units` classes?

Reply:
I think it should be " ...a units... " because the "u" in units makes a consonant sound and not a vowel sound.

Reply:
it's not the "a" vs "an" that I'm talking about - I agree that "a unit" is better than "an unit".  It's the plural vs singular of "unit" meaning different things: we use the plural of "unit" to mean something other than more than one unit, we use it to mean the `libCellML::Units` items.  I think we should call our `libCellML::Units` class something like `CompoundUnit` so that the distinction is clearer and easier to write/talk about.

Reply:
That was really a comment to @agarny on the English not a comment on the use of the term "units" in a singular manner.

Reply:
> I think it should be " ...a units... " because the "u" in units makes a consonant sound and not a vowel sound.

Yes, that's what I meant and my point since right now the message reads "... an units..."

Reply:
> I think it should be " ...a units... " because the "u" in units makes a consonant sound and not a vowel sound.

This still needs to be corrected.

Reply:
Sorry missed the first use was only looking at the second.